cabal-install/src/Distribution/Client/SetupWrapper.hs

```diff
@@ -875,6 +1056,11 @@ getExternalSetupMethod verbosity options pkg bt = do
         | otherwise
         -> error "buildTypeScript Hooks with Cabal < 3.13"
       Custom -> error "buildTypeScript Custom"
+      -- TODO: should all of these include {-# LANGUAGE NoImplicitPrelude #-}?
+      -- What happens if there is no base dependency declared in the Cabal file?
```

Comment:
It sounds like a good idea to include NoImplicitPrelude. In fact, I've run against this I think:).

Reply:
I did this and put it up as #9988.